test/test-ContentTypeMatcher.cc

```diff
@@ -0,0 +1,34 @@
+#include "ContentTypeMatcher.h"
+
+#include <catch2/catch.hpp>
+
+using state = parser::mediatype::state;
+using namespace drafter;
+
+TEST_CASE("test JSON comparator") {
+
+    JSONContentTypeComparator json;
+
+    REQUIRE(json == state{"application", "json" });
+    REQUIRE(json == state{"application", "hal", "json" });
+    REQUIRE(json == state{"application", "json", {}, {{"key", "value"}}});
+
+    REQUIRE_FALSE(json == state{"dummy", "hal", "json" });
+
+}
+
+TEST_CASE("test matchers") {
```

Comment:
You are right i wanted to discuss it. 
Probably simplest solution is hold parsed info in lower case.
Or is preferred solution to hold it untouched and take care just while compare?

Reply:
I think it depends, if we ever want to serialize the parsed media type (to place it in the parse result), we should probably keep it intact and handle case during compare.

If we will never do this, then I think storing as lowercase is acceptable.

Reply:
Currently we do not use it, but In future we want to use parsers in APIB grammar, right?

So probably it will be better just handle it while compare as you suggest